openapi.yaml

```diff
@@ -0,0 +1,1252 @@
+
+        and the consumer with the reservation has not marked the message via 
+        `/queue.complete` or `queue.defer`
+
+
+        If a message is not acknowledge via `/queue.complete` within the
+        `reserve_deadline` period it will be offered to another consumer.
+
+
+        ### Timeout Semantics
+
+        The call will return when a reservation is successful or when
+        `request_timeout` has been reached. If the call returns with a 454
+        `Retry Request` response code, this means there were no items in the
+        queue and the `request_timeout` was reached. In this case the client
+        should retry the reserve request.
+
+
+        The client should NOT cancel the request before `request_timeout` has
+        been reached. If the call fails to return after `request_timeout` is
+        reached the client can safely cancel the request.
+
+
+      operationId: post-v1-queue.reserve
+      tags:
+        - Querator V1
+      parameters: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/QueueReserveRequest'
+            example:
+              queue_name: queue-name
+              batch_size: 1000
+              client_id: client-01
+              request_timeout: 30s
+      responses:
+        '200':
+          description: ''
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/QueueReserveResponse'
+              examples:
+                '1':
+                  summary: Example 1
+                  value:
+                    items:
+                      - encoding: application/json
+                        kind: webhook-v2
+                        reference: account-1234
+                        id: 2m75RTp9PBx69hw1Q7mjoB0F73Q
+                        reserve_deadline: '2024-07-02T20:50:49.366215Z'
+                        bytes: >-
+                          R29vZCBuZXdzIGV2ZXJ5b25lLCBteSBJUSB0ZXN0IGNhbWUgYmFjayBuZWdhdGl2ZQ==
+                    queue_name: webhooks
+                    partition: 0
+          headers: {}
+      security: []
+  /v1/queue.defer:
+    post:
+      summary: Defer a message to be retried
+      deprecated: false
+      description: >
+        Defer an item to be processed again at a later date/time. Once the
+        `enqueue_at` time is reached, querator will enqueue this item into the
+        specified queue and the `attempted` counter for this item will increment
+        by 1. 
+
+
+        If the item should not be retried, but failed immediately. Set `dead:
+        true` and the item will be placed in the dead letter queue regardless of
+        the number of attempts left on the message.
+
+
+        If you wish to fail a message such that it is deleted from the queue and
+        never offered to another consumer, nor placed into the dead letter
+        queue, then call `/queue.complete` instead.
```

Comment:
This one caught me briefly. I was wondering why `dead: true` was on `/queue.defer` and not `/queue.complete`? I figured it depended on the intention for processing dead letter queues but was curious 👀

Reply:
Calls to `/queue.complete` will never result in moving an item to the dead letter queue. But there are some situations where you want to fail an item without having to wait for all attempts to be exhausted, so I added `dead: true` to the `/queue.defer` call